packages/web3-providers-http/src/index.js

```diff
@@ -112,6 +112,16 @@ HttpProvider.prototype.send = function (payload, callback) {
         }
     };
 
+	//since XHR2._onHttpRequestError swallows the initial request error, we need to get it from the underlying request
+	request.addEventListener('loadstart', function() {
+		var clientRequest = request._request;
+		if (clientRequest) {
+			clientRequest.on('error', function (error) {
+				callback(error || errors.RequestFailed());
```

Comment:
```suggestion
				callback(error || new Error('HTTP request failed'));
```

If `RequestFailed` [link](https://github.com/ChainSafe/web3.js/pull/3947/files#diff-0ce97ab24cb3318dbe5a48699a03e2c31e13a82418908e1bd3e29e91cebb7b48R47) is only used here, could we just return the error instead of calling a function?

Reply:
@spacesailor24 absolutely.